src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -542,6 +542,62 @@ theorem eq_zero_or_eq_one (a : BitVec 1) : a = 0#1 ∨ a = 1#1 := by
 theorem toInt_zero {w : Nat} : (0#w).toInt = 0 := by
   simp [BitVec.toInt, show 0 < 2^w by exact Nat.two_pow_pos w]
 
+theorem toInt_lt {w : Nat} (x : BitVec w) : x.toInt < 2 ^ (w - 1) := by
+  simp only [BitVec.toInt, Nat.cast_pow]
+  by_cases hw : w = 0
```

Comment:
Would using `cases` or `rcases` be easier here?

Reply:
it should, will give it a try